src/server.cpp

```diff
@@ -327,8 +327,10 @@ Server::~Server()
 	}
 
 	// Delete things in the reverse order of creation
-	delete m_emerge;
 	delete m_env;
+	// ServerEnvironment::deactivateFarObjects
+	// requires EmergeManager to be alive
+	delete m_emerge;
```

Comment:
Note to reviewers: This fixes a segfault in `deactivateFarObjects` where `EmergeManager` was required in a later stage. Leaving the world now produces warnings, but that's far better than a crash.